### ISSUE TITLE
configuration args로 담는 방법들 정리

### ISSUE THREAD

**Issue by sallyy1** [sallyy1/NLU-NLP#17]
# 1. argparse 사용
- import argparse
## 1-1. argparse 터미널, 도커에서 활용

# 1-2. argparse 노트북에서 사용하기 위한 팁
- 두 줄 입력 필수
```
parser = argparse.ArgumentParser()
args = parser.parse_args("") # ipynb - ""
```
- 적용 예: 지난 스터디 공유 (https://colab.research.google.com/drive/1W-SWMyCjWaHH0eJWZqjvAk2gJdUd03Uf#scrollTo=iNI9xj1d23cs)

# 2. OrderedDict 사용
- from collections import OrderedDict 로 임포트 해줘야 한다.
- 딕셔너리처럼 []로 표현된다.

- 적용 예
<img width="603" alt="스크린샷 2021-02-13 오후 1 44 17" src="https://user-images.githubusercontent.com/70500841/107841831-d2777980-6e01-11eb-991a-24e9dad354b5.png">

- print(args) 결과
OrderedDict([('SEED', 42),
             ('NUM_FAKE', 86002),
             ('NUM_REAL', 13453),
             ('EPOCH', 10),
             ('init_lr', 0.0003),
             ('IMAGE_SIZE', [160, 160, 3]),
             ('CROP_FACTOR', 0.9),
             ('AUTO', -1),
             ('SHUFFLE_BUFFER_SIZE', 1000000),
             ('TEST_SIZE', 0.2),
             ('IMAGE_INPUT_NAME', 'image'),
             ('LABEL_INPUT_NAME', 'label'),
             ('BACKBONE_MODEL',
              <function tensorflow.python.keras.applications.xception.Xception>),
             ('adv_multiplier', 0.2),
             ('adv_step_size', 0.001),
             ('adv_grad_norm', <NormType.L2: 'l2'>),
             ('GLOBAL_BATCH_SIZE', 64)])

- ref: https://dacon.io/competitions/official/235655/codeshare/1809?page=1&dtype=recent&ptype=pub

# 3. SimpleNamespace 사용
- from types import SimpleNamespace 로 임포트 해줘야 한다.
- 온점(.)으로 표현 가능하다. (= bash 상과 동일)

- 적용 예
<img width="658" alt="스크린샷 2021-02-12 오후 11 05 10" src="https://user-images.githubusercontent.com/70500841/107841692-61839200-6e00-11eb-8dc2-ceb7f186eb5a.png">

- print(args) 결과
namespace(beta1=0.5, beta2=0.999, c2_dim=8, c_dim=5, d_conv_dim=64, d_lr=0.0001, d_repeat_num=6, dataset='CelebA', g_conv_dim=64, g_lr=0.0001, g_repeat_num=6, image_size=256, mode='test', model_save_dir='stargan_celeba_256/models', num_workers=1, selected_attrs=['Black_Hair', 'Blond_Hair', 'Brown_Hair', 'Male', 'Young'], test_iters=200000)

- ref: https://colab.research.google.com/github/ndb796/Deep-Learning-Paper-Review-and-Practice/blob/master/code_practices/StarGAN_Tutorial.ipynb
